### PR TITLE
[BugFix] Qwen2.5-omni supress end token and won't stop

### PR DESCRIPTION
**Pull request by yinpeiqi**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose

In the current implementation, `bad_word_processor` will set the logits to -1e9 when encounter supress tokens, to avoid the vocab size mismatch. We found that the end token also will be set to -1e9, thus the program won't stop until reach max length.

## Test Plan

Run end2end.py

## Test Result

Before:

[0_8990c4c2-dede-49cf-95b1-5cb93ef0ad05.txt](https://github.com/user-attachments/files/24589945/0_8990c4c2-dede-49cf-95b1-5cb93ef0ad05.txt)
[output_0_8990c4c2-dede-49cf-95b1-5cb93ef0ad05.wav](https://github.com/user-attachments/files/24589952/output_0_8990c4c2-dede-49cf-95b1-5cb93ef0ad05.wav)

After:

[0_163825c2-5c08-4fd1-89f1-f4ac9668ea26.txt](https://github.com/user-attachments/files/24589995/0_163825c2-5c08-4fd1-89f1-f4ac9668ea26.txt)
[output_0_163825c2-5c08-4fd1-89f1-f4ac9668ea26.wav](https://github.com/user-attachments/files/24589998/output_0_163825c2-5c08-4fd1-89f1-f4ac9668ea26.wav)

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

@Bounty-hunter @hsliuustc0106

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
@yenuo26 how can we use tests to catch such error?

**Comment by yenuo26** [comment]
> @yenuo26 how can we use tests to catch such error?

we use whisper in test case to convert speech to text. If the audio contains long segments of silence, it may lead to a discrepancy between the recognized text and the inferred text, ultimately causing the test case to fail.
like this pr: #720
